pkg/destinations/redshift/query.go

```diff
@@ -0,0 +1,115 @@
+package redshift
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+)
+
+func (s *RedshiftServer) QueryJSON(query string, writer io.Writer) error {
+
+	rows, err := s.conn.Query(query)
+	if err != nil {
+		return fmt.Errorf("failed to execute query: %v", err)
+	}
+	defer rows.Close()
+
+	columns, err := rows.Columns()
+	if err != nil {
+		return fmt.Errorf("failed to get column names: %v", err)
+	}
+
+	values := make([]interface{}, len(columns))
+	valuePtrs := make([]interface{}, len(columns))
+	for i := range columns {
+		valuePtrs[i] = &values[i]
+	}
+
+	
+	jsonObjects := []map[string]interface{}{}
+
+	for rows.Next() {
+		err := rows.Scan(valuePtrs...)
+		if err != nil {
+			return fmt.Errorf("failed to scan row values: %v", err)
+		}
+
+		// Populate the JSON object with column values
+		jsonObject := make(map[string]interface{})
+		for i, column := range columns {
+			jsonObject[column] = values[i]
+		}
+
+		jsonObjects = append(jsonObjects, jsonObject)
```

Comment:
Do not load the entire dataset in memory. Instead, stream data as it is generated. See Step 2 of #117:

> Data should be streamed to the writer and avoid allocations as much as possible. Do not buffer all data into memory before returning to the client. Do not generate the entire JSON/CSV payload in memory.

Reply:
Noted

Reply:
Done